test/unit/kaocha/watch_test.clj

```diff
@@ -186,3 +187,57 @@
+        config (-> (config/load-config config-file)
+                   (assoc-in [:kaocha/cli-options :config-file] (str config-file))
+                   (update :kaocha/tests (fn [suites]
+                                           (let [base-suite (assoc (first suites)
+                                                                  :kaocha/source-paths []
+                                                                  :kaocha/test-paths [(str test-dir)])]
+                                             [(assoc base-suite :kaocha.testable/id :first-suite)
+                                              (assoc base-suite :kaocha.testable/id :second-suite)]))))
+        _ (spit (str config-file) (pr-str config))
+        spit-good-test #(integration/spit-file m "test/bar_test.clj" (str "(ns bar-test (:require [clojure.test :refer :all])) (deftest good-test (is true))"))
+        spit-bad-test #(integration/spit-file m "test/bar_test.clj" (str "(ns bar-test) (throw (Exception. \"Intentional compilation error\"))"))
+
+        dbg (bound-fn* prn)
+        _ (dbg "before")
+        _ (spit-good-test)
+        exit (integration/interactive-process m [":second-suite" "--watch"]
+               (try
+                 (dbg "first lines")
+                 (integration/expect-lines
+                   ["[(.)]"
+                    "1 tests, 1 assertions, 0 failures."
+                    ""])
+                 (spit-bad-test)
+                 (dbg "after bad test")
+                 (integration/expect-lines
+                   ["[watch] Reloading #{bar-test}"
+                    "[E]"
+                    ""
+                    "ERROR in second-suite (bar_test.clj:1)"
+                    "Failed reloading bar-test:"])
+                 (dbg "compiler exception")
+                 (integration/next-line-matches
+                   #{"Exception: clojure.lang.Compiler$CompilerException: Syntax error macroexpanding at (bar_test.clj:1:15)."
+                     "Exception: clojure.lang.Compiler$CompilerException: Syntax error compiling at (bar_test.clj:1:15)."
+                     "Exception: clojure.lang.Compiler$CompilerException: java.lang.Exception: Intentional compilation error, compiling:(bar_test.clj:1:15)"})
+                 (dbg "big trace")
+                 (integration/read-until #{"1 tests, 1 assertions, 1 errors, 0 failures."})
+                 (integration/expect-lines
+                   [""
+                    "[watch] Error reloading, all tests skipped."])
+                 ;; fix the compilation error...
+                 (spit-good-test)
+                 (dbg "after good-test")
+                 (integration/expect-lines
+                   ["[watch] Reloading #{bar-test}"
+                    "[(.)]"
+                    "1 tests, 1 assertions, 0 failures."])
+                 (finally
+                   ;; unsure how to exit process via (println) ... eg., how to enter ^C ?
```

Comment:
What you have seems to work, but maybe if *process* were a map in an atom, you could have a `:process` key containing an actual `Process` and `:continue` key telling it whether to keep going? I think this might make `*interactive-process*` a little tidier, too.

We can also save this reworking for a later PR since I don't want to hold up this bug fix too much longer.

Reply:
I pasted your suggestion in this comment for later.